### PR TITLE
feat(runtime): use committee origin for pallet asset-index

### LINKED ISSUE #268: Check origins for all pallets
### pallet-xcm

+ [ ] SendXcmOrigin -  `EnsureXcmOrigin<Origin, LocalOriginToLocation>`

### cumulus_pallet_dmp_queue

+ [ ] ExecuteOverweightOrigin - `frame_system::EnsureRoot<AccountId>`

### pallet_collator_selection

+ [ ] UpdateOrigin - `EnsureApprovedByCommittee`

### pallet_local_treasury

+ [ ] AdminOrigin - `frame_system::EnsureRoot<AccountId>`

### pallet_saft_registry

+ [ ] AdminOrigin - `frame_system::EnsureSigned<AccountId>`

### pallet_committee

+ [ ] ProposalSubmissionOrigin - `EnsureSigned<AccountId>`
+ [ ] ProposalExecutionOrigin - `EnsureMember<Self>`
+ [ ] ApprovedByCommitteeOrigin - `EnsureApprovedByCommittee`

### pallet_price_feed

+ [ ] AdminOrigin - `frame_system::EnsureRoot<AccountId>`

### pallet_asset_index

+ [ ] AdminOrigin - `frame_system::EnsureSigned<AccountId>`

### pallet_remote_asset_manager

+ [ ] AdminOrigin - `frame_system::EnsureSigned<AccountId>`

## Other information and links

- #104 
- #140

### LINKED ISSUE #302: Supports creating feed in price-feed in feature runtime-benchmarks
## Issue summary

The `deposit` call of asset-index requires exists feed, which we can not add in its benchmarks directly, needs to support this in `runtime-benchmarks` feature

## Other information and links
-

### PR DESCRIPTION
**Pull request by clearloop**
## Changes

-

## Tests

```
CI
```

## Issues

- Closes #302 
- #268

### CODE REVIEW

**Comment by clearloop** [comment]
This PR simply add an interface for creating feed in pallet price-feed

The benchmarks of asset-index is still broken  with:

```
Error:                                                                                                                                 
* Pallet: AssetIndex                                                                                                                   
* Benchmark: deposit                                                                                                                   
* Lowest_range_values: []                                                                                                              
* Highest_range_values: []                                                                                                             
* Steps: []                                                                                                                            
* Repeat: 1                                                                                                                            
* Verify: true                                                                                                                         
* Error message: An overflow would occur 
```

Which is from the pallet-chainlink-feed, tried `submit` round in the `chainlink-feed` node-template directly, not work as well

File: pallets/price-feed/src/lib.rs [hunk 1]
```diff
@@ -260,6 +265,45 @@ pub mod pallet {
...
+			pallet_chainlink_feed::Pallet::<T>::submit(
+				<frame_system::Origin<T>>::Signed(caller.clone()).into(),
+				feed_id,
+				1_u32.into(),
+				42.into(),
+			)?;
+			Ok(().into())
+		}
+	}
+
```
Comment (mattsse) [inline pallets/price-feed/src/lib.rs:306]: This is essentially a mock?
I'm rather against adding this. Can you extend on why this is necessary, we should fix this upstream if this is a problem of the chainlink pallet.

**Review by mattsse — CHANGES_REQUESTED** [review]
I don't really understand the price feed benchmark extensions and I don't think we should add this like that because the `PriceFeedBenchmarks` is very invasive. But maybe I didn't get the problem.

**Comment by mattsse** [comment]
Also there is already the `Feedbuilder` we use for price feed tests https://github.com/ChainSafe/PINT/blob/bb27364ebe9cc7021860630aa7502e80bc6b7999/pallets/price-feed/src/mock.rs
can't we use that instead?

**Comment by clearloop** [comment]
> Also there is already the `Feedbuilder` we use for price feed tests https://github.com/ChainSafe/PINT/blob/bb27364ebe9cc7021860630aa7502e80bc6b7999/pallets/price-feed/src/mock.rs
> can't we use that instead?

The dependencies benchmarks depend on are not as same as the tests, they are using the real runtimes, and whatever we have `MockPriceFeed` or `Feedbuilder`, the benchmarks don't care about these, they only work for tests ( including the tests in benchmarks ), but not work for benchmarks, 

So if we want to generate weights for extrinsics like `deposit` in pallet asset-index, we need to `create_feed` and `submit` a round in the pallet asset-index, it used to be workable because we had a cache for the price of a new asset before

File: pallets/price-feed/src/lib.rs (hunk 1 above)
Comment (clearloop) [reply pallets/price-feed/src/lib.rs:306]: Yep, I'm thinking of fixing it in the chainlink repo and open another issue for this

**Comment by mattsse** [comment]
gotcha,
this is probably solvable because the `PriceFeed` pallet is also `chainlink_feed_pallet::Pallet` so you should be able to access that, and we need to enable this in the runtime section as well?
this could work?

**Comment by clearloop** [comment]
> gotcha,
> this is probably solvable because the `PriceFeed` pallet is also `chainlink_feed_pallet::Pallet` so you should be able to access that, and we need to enable this in the runtime section as well?
> this could work?

Yep, and this `PriceFeedBenchmarks` is for this creating feed requirement, tried to re-generate weights for all pallets, both `asset-index` and `saft-registry` has the overflow problem mentioned above, not certain about why it happens currently now

**Comment by mattsse** [comment]
Okay seems we need to track this down.
Can you please split the committee origin and create a separate PR for this?

**Comment by clearloop** [comment]
> Okay seems we need to track this down.
> Can you please split the committee origin and create a separate PR for this?

Yep, in #305 now =.=

**Review by mattsse — APPROVED** [review]
lgtm

**Pull request merged**